### DISCUSSION TITLE
Removing an event from a calendar?

### DISCUSSION THREAD

**Discussion by VaguelyOnline** [spatie/icalendar-generator#53] [Q&A]
Hi,

Great package - thanks very much! Is there a way to create an ics that will remove an item from the users' calendar? I've tried creating a new Event with the same UUID as the initial invite, and have set the status of the event to cancelled ($event->status(EventStatus::cancelled()); ) but with Gmail, it just shows up as a new item to be added to the calendar, rather than prompting the user to remove the existing one.

Is this functionality possible and I've just missed it? Thanks in advance.

**Comment by rubenvanassche** [comment]
Hi @nwoodhouse,

Actually I have no idea how to solve such an issue, the process you described seems correct to me and I would expect that Google to cancel the event. Maybe this only works if you have a calendar url that streams the calendar from a server? Not sure about that but I think Google might use this as a security practice not to overwrite earlier events if they were added manually?

**Reply by VaguelyOnline** [reply]
Hi @rubenvanassche 

Thanks for the reply and for giving it some thought :-).

Comparing it to a cancel created directly from google calendar - I think it may have to do with the METHOD field not being set in the ICAL created by this package - ring any bells?

**Reply by rubenvanassche** [reply]
Hi @nwoodhouse,

It indeed seems to be the method field, which isn't part of the iCalendar standard but more information about can be found here: https://datatracker.ietf.org/doc/html/rfc2446#section-3.2

Could you try to append(https://github.com/spatie/icalendar-generator#appending-properties) such a METHOD property and see if that works?

If that's the case maybe we should add it to the package by default.

Ruben

**Comment by kovaliman** [comment]
Hi @nwoodhouse @rubenvanassche As I can see no answer for this issue. I tried now this and everything works fine.
To automaticly create event in google calendar in ics attachment, set property attendee in Event class and in Calendar class set appendProperty(TextProperty::create('METHOD', 'REQUEST'))
To delete that event from calendar just set in Calendar class appendProperty(TextProperty::create('METHOD', 'CANCEL')) with same event UID, ofcourse.
I hope this can help somebody. And thanks for the great package.

**Reply by optmsp** [reply]
Awesome tip, thank you!

**Reply by rubenvanassche** [reply]
That's cool!
